packages/util/src/signature.ts

```diff
@@ -103,41 +133,47 @@ export const toCompactSig = function (v: bigint, r: Buffer, s: Buffer, chainId?:
 
 /**
  * Convert signature format of the `eth_sign` RPC method to signature parameters
- *
  * NOTE: For an extracted `v` value < 27 (see Geth bug https://github.com/ethereum/go-ethereum/issues/2053)
  * `v + 27` is returned for the `v` value
  * NOTE: After EIP1559, `v` could be `0` or `1` but this function assumes
  * it's a signed message (EIP-191 or EIP-712) adding `27` at the end. Remove if needed.
  */
-export const fromRpcSig = function (sig: string): ECDSASignature {
+export const fromRpcSig = function (sig: string, chainId?: bigint): ECDSASignature {
   const buf: Buffer = toBuffer(sig)
 
   let r: Buffer
   let s: Buffer
   let v: bigint
+  let recovery: bigint
   if (buf.length >= 65) {
     r = buf.slice(0, 32)
     s = buf.slice(32, 64)
     v = bufferToBigInt(buf.slice(64))
+    if (v < 27) {
+      v = v + BigInt(27)
+    }
+    recovery = calculateSigRecovery(v, chainId)
   } else if (buf.length === 64) {
     // Compact Signature Representation (https://eips.ethereum.org/EIPS/eip-2098)
     r = buf.slice(0, 32)
     s = buf.slice(32, 64)
     v = BigInt(bufferToInt(buf.slice(32, 33)) >> 7)
+    if (v < 27) {
+      v = v + BigInt(27)
+    }
+    recovery = calculateSigRecovery(v, chainId)
     s[0] &= 0x7f
   } else {
     throw new Error('Invalid signature length')
   }
 
   // support both versions of `eth_sign` responses
-  if (v < 27) {
-    v = v + BigInt(27)
-  }
```

Comment:
It's not necessary to double this code above since `recovery` isn't used up till here at all, so recovery calculation could be just added here, or am I missing something?